### PR TITLE
[stable25] fix: don't emit Hooks when hookpaths are empty

### PR DESCRIPTION
**Pull request by miaulalala**
Manual backport of https://github.com/nextcloud/server/pull/39481

### CODE REVIEW

**Comment by miaulalala** [comment]
linter is giving me 905 files lol

**Comment by miaulalala** [comment]
Failing test is unrelated (acceptance-users) :

```
--- Failed scenarios:
229 |  
230 | /drone/src/tests/acceptance/features/users.feature:25
231
```

**Review by max-nextcloud — APPROVED** [review]

**Review by icewind1991 — APPROVED** [review]

**Comment by max-nextcloud** [comment]
php-cs check passed for another PR - so i will try and rebase this so it runs again.

File: lib/private/Files/View.php [hunk 1]
```diff
@@ -819,19 +823,19 @@ public function rename($path1, $path2) {
...
-							// moving a file/folder within the same mount point
+						// moving a file/folder within the same mount point
 						} elseif ($storage1 === $storage2) {
 							if ($storage1) {
 								$result = $storage1->rename($internalPath1, $internalPath2);
 							} else {
 								$result = false;
 							}
-							// moving a file/folder between storages (from $storage1 to $storage2)
+						// moving a file/folder between storages (from $storage1 to $storage2)
```
Comment (max-nextcloud) [inline lib/private/Files/View.php:833]: Looks like the linter does not like this.
I'll move the comments into the respective block.

**Comment by miaulalala** [comment]
Thanks @max-nextcloud !!!

Drone failure unrelated:

```
--- Failed scenarios:
--
229 |  
230 | /drone/src/tests/acceptance/features/users.feature:25

```

**Comment by max-nextcloud** [comment]
Now if only someone could merge this despite the failing CI
@ChristophWurst @juliushaertl 

# :pleading_face: 

:wink:

**Pull request merged**